### DISCUSSION TITLE
[Nakamoto] Should Stacks support Schnorr Signature Scheme Transaction Types

### DISCUSSION THREAD

**Discussion by AshtonStephens** [stacksgov/sips#160] [Ideas]
This is a discussion post around answering the question "does a new transaction auth variant require a new address type?" or for SIP-021's specific purposes this would be "Should we add a Schnorr signature scheme transaction type to the blockchain?"

**Relevant Draft SIP:** SIP-021 https://github.com/stacksgov/sips/pull/155

If it's a new transaction, it has to be in the SIP. But that's only if we go that route.

**Comment by AshtonStephens** [comment]
There are some additional questions to the implications of a new address type if that's the chosen route. For instance, wallets *may* want to add support for that address type.

**Comment by jcnelson** [comment]
Pros:

* **No magical signature verification**.  If there's first-class support for Schnorr signatures in transactions, then it's apparent how to validate the signature.  The Stacks node does not need to be involved in any special way

* **Scalable weighted multisig**.  A single Schnorr signature can represent a threshold signature of many, many entities with different voting weights, which is useful in its own right.  Moreover, it's more private than traditional multisig because the individual signers are not disclosed in the transaction.  This is useful for things like DAOs and decentralized stacking pools.

Cons:

* We need a new `TransactionAuth` variant

* We probably need a new address type (besides `SP`, `SM` on mainnet and `ST`, `SN` on testnet), and wallets will need to support it

**Comment by andrerserrano** [comment]
It seems like this decision is simple:
- Is there user demand for a new address type? No. 
- Is it required for the Stacks blockchain to process Schnorr signatures? No. 

My understanding is that this can be done with the current SP transaction types by just indicating a Schnorr signature. Furthermore, the amount of unknowns involved with creating a new address type, engineering resources, and scope of work for wallets to integrate and make compatible are likely not justified.

**Reply by jcnelson** [reply]
> Is there user demand for a new address type? No.

There is, however, demand for order-independent multisig, there is definitely demand for stacking pools (which require threshold signatures to work at scale), and no one would say no to an efficient way to store and validate many signatures.

> Is it required for the Stacks blockchain to process Schnorr signatures?

This is more of an unknown.  We'll know for sure when we try to implement it.

**Reply by andrerserrano** [reply]
Agree, having an efficient way to validate threshold signatures at scale is beneficial. We just need to weigh the trade offs in order to ship in a reasonable timeframe and currently there is not a strong argument to add a new address. If we try to implement and realize its not possible, then we can always revisit.
